xray/core/dataset.py

```diff
@@ -632,6 +638,9 @@ def __setitem__(self, key, value):
         if utils.is_dict_like(key):
             raise NotImplementedError('cannot yet use a dictionary as a key '
                                       'to set Dataset values')
+
+        validate_dataarray_name(key)
```

Comment:
Okay, I get it now.  For some reason, I didn't see that `update` eventually called `_update_vars_and_coords`.